src/details-element-polyfill/polyfill.coffee.erb

```diff
@@ -56,11 +56,37 @@ polyfillProperties = ->
           removeAttribute.call(this, name)
 
 polyfillToggle = ->
-  onTogglingClick (element) ->
+  onTogglingTrigger (element) ->
+    summary = element.querySelector "summary"
     if element.hasAttribute("open")
       element.removeAttribute("open")
+      summary.setAttribute("aria-expanded", false)
     else
       element.setAttribute("open", "")
+      summary.setAttribute("aria-expanded", true)
+
+polyfillFocusAndARIA = ->
+  summaries = document.querySelectorAll("summary")
+  for summary in summaries
+    addAttributesForSummary(summary)
+  if MutationObserver?
+    observer = new MutationObserver (mutations) ->
+      for {addedNodes} in mutations
+        for target in addedNodes
+          if target.tagName is "DETAILS"
+            summary = target.querySelector "summary"
+            addAttributesForSummary(summary) if summary
+    observer.observe(document.documentElement, attributes: true, subtree: true, childList: true)
+  else
+    document.addEventListener "DOMNodeInserted", (event) ->
+      if event.target.tagName is "SUMMARY"
+        addAttributesForSummary event.target
+
+addAttributesForSummary = (summary) ->
+  details = findClosestElementWithTagName(summary, "DETAILS")
+  summary.setAttribute "tabindex", "0"
+  summary.setAttribute "aria-expanded", details.hasAttribute("open")
+  summary.setAttribute "role", "button"
```

Comment:
Should probably avoid overwriting these attributes if they're present initially:
```coffee
summary.setAttribute("tabindex", "0") unless summary.hasAttribute("tabindex")
...
```